docs/env.md

```diff
@@ -0,0 +1,78 @@
+# Environment Variables
+
+When running fwupd reads some variables from your environment and changes some
+behavior. This might be useful for debugging, or to make fwupd run somewhere
+with a non-standard filesystem layout.
+
+## fwupdmgr and fwupdtool
+
+* `DISABLE_SSL_STRICT` disables strict SSL certificate checking, which may make
+  downloading files work when using some antisocial corporate firewalls.
+* `FWUPD_CURL_VERBOSE` shows more information when downloading files
+* `FWUPD_DEVICE_TESTS_BASE_URI` sets the base URI when downloading firmware for the device-tests
+* `FWUPD_SUPPORTED` overrides the `-Dsupported_build` meson option at runtime
+* `FWUPD_VERBOSE` is set when running `--verbose`
+* `FWUPD_XMLB_VERBOSE` can be set to show Xmlb silo regeneration
+
+## Self Tests
+
+* `CI_NETWORK` if CI is running with network access
+* `TPM_SERVER_RUNNING` if an emulated TPM is running
+
+## Shared libfwupdplugin
+
+* `FU_HID_DEVICE_VERBOSE` shows HID traffic
+* `FU_SREC_FIRMWARE_VERBOSE` shows more information about parsing
+* `FU_UDEV_DEVICE_DEBUG` shows more information about UDEV devices, including parents
+* `FU_USB_DEVICE_DEBUG` shows more information about USB devices
+* `FWUPD_DEVICE_LIST_VERBOSE` display devices being added and removed from the list
+* `FWUPD_PROBE_VERBOSE` dump the detected devices to the console, even if not supported by fwupd
+
+## Plugins
+
+Most plugins read a plugin-specific key to increase verbosity, e.g. `FWUPD_DELL_VERBOSE`.
```

Comment:
I think explaining how the option for plugin verbose sets this makes sense

Reply:
Added